cmake/BuildFlags.cmake

```diff
@@ -64,8 +64,16 @@ if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" OR CMAKE_CXX_COMPILER_ID STREQUAL "MSVC"
     set(SYCL_KERNEL_OPTIONS ${SYCL_KERNEL_OPTIONS} -D_GLIBCXX_USE_CXX11_ABI=${GLIBCXX_USE_CXX11_ABI})
   endif()
 
-  set(SYCL_KERNEL_OPTIONS ${SYCL_KERNEL_OPTIONS} -fsycl-fp64-conv-emu)
-  
+  if(SYCL_COMPILER_VERSION)
+    set(MIN_DATE_VERSION "20240604")
+    if(SYCL_COMPILER_VERSION GREATER_EQUAL MIN_DATE_VERSION)
+      message(STATUS "Enable Double Emulation Feature")
+      set(SYCL_KERNEL_OPTIONS ${SYCL_KERNEL_OPTIONS} -fsycl-fp64-conv-emu)
+    else()
+      message(STATUS "SYCL Compiler Version ${SYCL_COMPILER_VERSION} is less than ${MIN_DATE_VERSION}. Disable Double Emulation Feature")
```

Comment:
I think `Warning` is better. And please enhance the log,
`"SYCL Compiler Version ${SYCL_COMPILER_VERSION} is less than ${MIN_DATE_VERSION}. Disable FP64 conversion emulation. There might be unknown behavior in some platforms that don't support FP64."`

Reply:
Done